config/src/main/java/com/typesafe/config/ConfigMemorySize.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config;
+
+/**
+ * An immutable class representing an amount of memory.  Use
+ * static factory methods such as {@link
+ * ConfigMemorySize#ofBytes(long)} to create instances.
+ */
+public final class ConfigMemorySize {
+    private final long bytes;
+
+    private ConfigMemorySize(long bytes) {
+        if (bytes < 0)
+            throw new IllegalArgumentException("Attempt to construct ConfigMemorySize with negative number: " + bytes);
+        this.bytes = bytes;
+    }
+
+    /** Constructs a ConfigMemorySize representing the given
+     * number of bytes.
+     */
+    public static ConfigMemorySize ofBytes(long bytes) {
+        return new ConfigMemorySize(bytes);
+    }
+
+    /** Gets the size in bytes.
+     */
+    public long toBytes() {
```

Comment:
yeah, I am thinking about that. The thing I'm uncertain about is that I'm not sure anyone ever actually wants to do that. Generally you want the input (the config) to have the other units, but then you end up making it bytes once you're in code. But perhaps there are exceptions. It ends up making the class quite a bit larger to add the other types.  Anyway I figured we can always add them later.